UI/NpcList/NpcList.gd

```diff
@@ -1,27 +1,124 @@
 extends Control
 
 onready var npcRow = load("res://UI/NpcList/NPCRow.tscn")
-onready var container = $VBoxContainer/ItemList/NpcListControld/ScrollContainer/HBoxContainer
+onready var container = $PanelContainer/VBoxC/ScrollC/VboxC2
+onready var nameButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Name
+onready var genderButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Gender
+onready var personalityButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/Personality
+onready var childrenButton = $PanelContainer/VBoxC/UpperPanel/HBoxContainer/ChildrenAmount
+var _nameBtnState: bool = true
+var _genderBtnState: bool = true
+var _personalityBtnState: bool = true
+var _childrenBtnState: bool = true
 
 
-func addRow(name: String, gender: String, personality: String, ID, children: int = 0):
+func addRow(name: String, gender: String, subbyStat: float, ID: String, children: int = 0):
 	var newRow = npcRow.instance()
 	container.add_child(newRow)
-	newRow.setName(name)
-	newRow.setGender(gender)
-	newRow.setPersonality(personality)
-	newRow.setNpcID(ID)
-	newRow.setChildrenAmount(children)
+	newRow.init(name, gender, subbyStat, ID, children)
 	newRow.connect("onForgetButtonPressed", self, "forgetNPC")
 
-
-func clearRows():
-	if(container != null):
-		for child in container.get_children():
-			container.remove_child(child)
-			child.queue_free()
-
 # add pop-up confirmation window
 func forgetNPC(ID):
 	GM.main.removeDynamicCharacter(ID)
 
+
+func _on_Name_pressed():
+	nameButton.pressed = true
+	genderButton.pressed = false
+	personalityButton.pressed = false
+	childrenButton.pressed = false
+	
+	var nodesSortedArr = container.get_children()
+	
+	if(_nameBtnState):
+		nodesSortedArr.sort_custom(self, "sortNameAscending")
+	else:
+		nodesSortedArr.sort_custom(self, "sortNameAscending")
+		nodesSortedArr.invert()
```

Comment:
I think you can just create a `sortNameDescending` function that's the opposite of the first one, means you will move less data around